low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -0,0 +1,382 @@
+        # exit as hard dependeny of SSPL
+
+        if consts.PRODUCT_NAME == 'LDR_R1':
+            if not os.path.exists(consts.REPLACEMENT_NODE_ENV_VAR_FILE):
+                CONSUL_PS = None
+                for proc in psutil.process_iter():
+                    if 'consul' in proc.name():
+                        CONSUL_PS = proc.pid
+                if not CONSUL_PS:
+                    raise SetupError(
+                            errno.EINVAL,
+                            "Consul is not running, exiting..")
+
+        # Get the types of server and storage we are currently running on and
+        # enable/disable sensor groups in the conf file accordingly.
+        if self.read_provisioner_config:
+            from cortx.sspl.bin.conf_based_sensors_enable import update_sensor_info
+            update_sensor_info()    
+            
+    def get_rabbitmq_cluster_nodes(self):
+        pout = None
+        if  self.rabbitmq_major_release == '3' and \
+            self.rabbitmq_maintenance_release == '8':
+
+            rmq_cluster_status_cmd = '/usr/sbin/rabbitmqctl cluster_status \
+                                    --formatter json'
+            output, error, returncode = SimpleProcess(rmq_cluster_status_cmd).run()
+            rabbitmq_cluster_status = json.loads(output)
+            if returncode:
+                raise SetupError(returncode, error)
+            running_nodes = rabbitmq_cluster_status['running_nodes']
+            for i, node in enumerate(running_nodes):
+                running_nodes[i] = node.replace('rabbit@', '')
+            pout = " ".join(running_nodes)
+        elif self.rabbitmq_version == '3.3.5':
+            rmq_cluster_status_cmd = "rabbitmqctl cluster_status | \
+                    grep running_nodes | cut -d '[' -f2 | cut -d ']' -f1 | \
+                    sed 's/rabbit@//g' | sed 's/,/, /g'"
+            output, error, returncode = SimpleProcess(rmq_cluster_status_cmd).run()
+            pout = output.replace("'", '').replace(' ', '')
+        else:
+            raise SetupError(
+                        errno.EINVAL, 
+                        "This RabbitMQ version : %s is not supported", 
+                        self.rabbitmq_version)
+
+        return pout
+
+    def get_cluster_running_nodes(self, msg_broker : str):
+        if(msg_broker == 'rabbitmq'):
+            return self.get_rabbitmq_cluster_nodes()
+        else:
+            raise SetupError(errno.EINVAL, 
+                        "Provided message broker '%s' is not supported", 
+                        msg_broker
+                        )       
+            
+
+    def process(self):
+        cmd = "config"
+        if(cmd == "config"):
+            self.config_sspl()
+        else:
+            raise SetupError(errno.EINVAL, "cmd val should be config")
+
+        # Get the version. Output can be 3.3.5 or 3.8.9 or in this format
+        rmq_cmd = "rpm -qi rabbitmq-server"
```

Comment:
check if we can use rpm validators

Reply:
No validators do not return anything, they just throw exception if pkg is is not installed.
Here we want to extract the version number from pkg info